src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -101,8 +98,15 @@ export class TransactionsHistoryMapper {
         )
           .filter(<T>(x: T | undefined): x is T => x != null)
           .flat();
-        transactions.push(...groupTransactions);
-        return transactions;
+
+        // If the current group is a follow-up from the previous page,
+        // or the group is empty, the date label shouldn't be added.
+        const isFollowUp = transactionGroup.timestamp == prevPageTimestamp;
```

Comment:
It is not strictly related, but this boolean was extracted to a variable to add some meaning and make it a bit more readable.